webapp/src/notification.tsx

```diff
@@ -0,0 +1,134 @@
+/// <reference path="../../typings/globals/react/index.d.ts" />
+/// <reference path="../../typings/globals/react-dom/index.d.ts" />
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import * as core from "./core";
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+export interface NotificationBannerProps extends ISettingsProps {
+    delayTime?: number; //milliseconds - delay before banner is shown
+    displayTime?: number; //milliseconds - duration of banner display
+    sleepTime?: number; //seconds - time to hide banner after it is dismissed
+    bannerType?: string;
+    content: JSX.Element;
+}
+
+export interface NotificationBannerState {
+    visible: boolean;
+}
+
+export class NotificationBanner extends data.Component<NotificationBannerProps, NotificationBannerState> {
+    delayTime: number;
+    doneSleeping: boolean;
+    bannerType: string;
+
+    //TODO need shouldComponentUpdate?
+
+    constructor(props: NotificationBannerProps) {
+        super(props);
+        this.delayTime = this.props.delayTime || 0;
+        this.doneSleeping = this.sleepDone();
+        this.bannerType = this.props.bannerType || "default";
+        this.state = {visible: false};
+
+        setTimeout(() => this.show(), this.delayTime);
```

Comment:
I think we should move this logic into a initialization function that runs in the app.tsx init phase. The constructor of a view should not be state changing.